framework/src/transfers/MultiAppMeshFunctionTransfer.C

```diff
@@ -219,7 +219,8 @@ MultiAppMeshFunctionTransfer::execute()
   {
     if (i_proc == processor_id())
       continue;
-    _communicator.send(i_proc, outgoing_points[i_proc]);
+    std::vector<Parallel::Request> reqs(n_processors());
```

Comment:
The idiom should be:
1. Create vector of requests
2. Use various requests to post sends
3. Loop over requests vector and call wait() on each one.

See, e.g. the code in system_subset_by_subdomain.C which does this.